id/id.go

```diff
@@ -0,0 +1,39 @@
+package id
+
+import (
+	"github.com/gofrs/uuid"
+)
+
+type AccessGroup struct {
+	ID     uuid.UUID
+	Scopes Scope
+}
+
+type Identity struct {
+	ID     uuid.UUID
+	Scopes Scope
+	Groups map[uuid.UUID]AccessGroup
+}
+
+func (i *Identity) GetIDs() map[uuid.UUID]struct{} {
```

Comment:
Maybe this should be called `GetGroupIDs` or similar as the Identity itself also has an idea